### PR TITLE
与香港城市大学合作项目Diff-MSR代码开源

### PR DESCRIPTION
**Pull request by wyhwhy**
接收于2024年WSDM的工作'Diff-MSR: A Diffusion Model Enhanced Paradigm for Cold-Start Multi-Scenario Recommendation'

### CODE REVIEW

**Review by vigo999 — APPROVED** [review]

File: research/huawei-noah/Diff-MSR/denoising_diffusion/continuous_time_gaussian_diffusion.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
+import math
+import torch
+from torch import sqrt
+from torch import nn, einsum
+import torch.nn.functional as F
+from torch.special import expm1
+
+from tqdm import tqdm
+from einops import rearrange, repeat, reduce
+from einops.layers.torch import Rearrange
```
Comment (geniuspatrick) [inline research/huawei-noah/Diff-MSR/denoising_diffusion/continuous_time_gaussian_diffusion.py:10]: torch?

Comment (wyhwhy) [reply research/huawei-noah/Diff-MSR/denoising_diffusion/continuous_time_gaussian_diffusion.py]: > torch?

Done now. Sorry that I uploaded unused files.

Comment (wyhwhy) [reply research/huawei-noah/Diff-MSR/denoising_diffusion/continuous_time_gaussian_diffusion.py]: @geniuspatrick There is only one file in the denoising_diffusion folder now. Please check and thanks.

**Comment by wyhwhy** [comment]
@geniuspatrick Sorry to bother. Is there any question for review?

**Pull request closed without merging**